### ISSUE TITLE
Restore single database from backup but preserve database on this host

### ISSUE THREAD

**Issue by stevemayster** [pgbackrest/pgbackrest#2209]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:
2.40

3. PostgreSQL version:
15

5. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
Centos 7.9

7. Did you install pgBackRest from source or from a package?
From source

9. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

10. Describe the issue:
Hello.
I want to know is it possible to restore one single database to new server and preserve existing database which running on this new server?
I install pgbackrest on new server and use this command

pgbackrest --config=/etc/pgbackrest.conf --log-level-console=info --stanza=mydb1-stanza --delta --db-include=tst --archive-mode=off restore

Logs:
2023-10-19 14:46:44.060 P00   INFO: repo1: restore backup set 20231019-124747F, recovery will start at 2023-10-19 12:50:30
2023-10-19 14:46:44.060 P00   INFO: nothing to filter - all user databases have been selected
2023-10-19 14:46:44.063 P00   INFO: remove invalid files/links/paths from '/u01/pgsql/15/data'
2023-10-19 14:50:50.294 P00   INFO: write updated /u01/pgsql/15/data/postgresql.auto.conf
2023-10-19 14:50:50.302 P00   INFO: restore global/pg_control (performed last to ensure aborted restores cannot be started)
2023-10-19 14:50:50.304 P00   INFO: restore size = 93.5GB, file total = 1525
2023-10-19 14:50:50.305 P00   INFO: restore command end: completed successfully (246562ms)

From docs i understand all technical database a restore too ( e.g. postgres and e.t.c.) but my goal to restore one single database from backup on new server and preserve old databases

**Comment by philrhurst** [comment]
In general, no it is not possible with pgBackRest to restore a single database on a server and preserve an existing database on that server.

pgBackRest does have the `--db-include` option which allows you to restore a single database, but the other database will be restored as "sparse, zeroed files" and not accessible.

- https://pgbackrest.org/command.html#command-restore
- https://pgbackrest.org/user-guide.html#restore/option-db-include

Now, if the server is running multiple clusters (i.e. Postgres is running with separate data directories and separate ports), then you may be able to achieve your objective. But there are other factors to consider.

**Comment by stevemayster** [comment]
Thank you, @philrhurst!
I have another question, maybe you know the answer.
I have master and standby on different sites.

I want to cross-backup it, like backup master in site A to backup server in site B and reversal.
From documentation i understand even i backup standby, pgbackrest connect to master to fetch wal-files.

My goal is to have a backups even one of sites failed,burned,etc.
Maybe you know some best practise about this?

Or better to open new issue?

**Comment by philrhurst** [comment]
Yes, even when backing up from a standby, pgBackRest will connect to the master to fetch several files (e.g. WAL, logs, statistics, etc). In my opinion, this provides a nice balance. Backing up from the standby allows the primary to function without the additional load. Connecting to the master to backup a minimal amount of data allows the backup to be comprehensive without overloading the primary.

- https://pgbackrest.org/user-guide.html#standby-backup

I am assuming that you are using pgBackRest for WAL archiving as well - if so, then even if one of your sites fail, then the Backrest repo will have the latest backup and all subsequent WAL files to perform a PITR to any point in the WAL stream. I would suggest building your DR strategy around native pgBackRest features: a separate Backrest repo to store WAL files and backups, using s3/s3-compatible object stores as additional repos, etc. pgBackRest recently added support for SFTP repositories. This is in addition to Azure/S3/S3-compatible/GCP object stores as well.

- https://pgbackrest.org/user-guide.html#multi-repo
- https://pgbackrest.org/user-guide.html#sftp-support

Hope this helps.
